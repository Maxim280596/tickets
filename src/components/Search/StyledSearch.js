import { Form } from "react-bootstrap";
import styled from "styled-components";

export const StyledForm = styled.form`
  font-size: 20px;
  box-shadow: 1px 1px 4px 1px #e0e0e0;
  display: flex;
  align-items: left;
  flex-direction: column;
  border-radius: 6px;
  background-color: white;
  width: 85%;
  padding-bottom: 10px;
  padding-top: 30px;
  padding-left: 0px;
  margin-left: 60px;
  

  & h5 {
    font-size:16px;
    margin-bottom: 10px;
    margin-left: 25px;
  }

  @media (max-width: 992px) {
    width: 100%;
    margin-left: 0px;
    margin-bottom: 20px;
  }
`;

export const StyledCheckbox = styled(Form.Check)`
  padding-left: 25px;
  padding-bottom: 10px;
  padding-top: 10px;
  width: 100%;
  &:hover {
    background-color: #f1fcff;
  }
`;
