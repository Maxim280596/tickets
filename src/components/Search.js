import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';

import {
  filteringOneStop,
  filteringAllStop,
  filteringNoStop,
  filteringThreeStop,
  filteringTwoStop,
} from '../src/redux/ducks/ticketsReducer';
import { StyledForm } from '../styled/StyledSearch';
import { SearchInput } from './SearchInput';

const Search = () => {
  const dispatch = useDispatch();

  const [state, setState] = useState({
    all: false,
    no: false,
    one: false,
    two: false,
    three: false,
  });

  let history = useHistory();

  const handleChange = (e) => {
    history.push(`/${e.target.name}`);
    setState({
      [e.target.name]: e.target.checked,
    });

    if (e.target.name === 'one') {
      dispatch(filteringOneStop());
    }
    if (e.target.name === 'two') {
      dispatch(filteringTwoStop());
    }
    if (e.target.name === 'three') {
      dispatch(filteringThreeStop());
    }
    if (e.target.name === 'all') {
      dispatch(filteringAllStop());
    }
    if (e.target.name === 'no') {
      dispatch(filteringNoStop());
    }
  };

  useEffect(() => {
    const { pathname } = history.location;
    setState({
      [pathname.substring(1)]: true,
    });
  }, []);

  return (
    <Router>
      <StyledForm>
        <div>
          <h5>КОЛИЧЕСТВО ПЕРЕСАДОК</h5>
          <SearchInput
            name="all"
            onChange={handleChange}
            id="all"
            inline
            label="Все"
            type="checkbox"
            checked={state.all}
          />
          <SearchInput
            name="no"
            onChange={handleChange}
            id="no"
            inline
            label="Без пересадок"
            type="checkbox"
            checked={state.no}
          />
          <SearchInput
            name="one"
            onChange={handleChange}
            id="one"
            inline
            label="1 пересадка"
            type="checkbox"
            checked={state.one}
          />
          <SearchInput
            name="two"
            onChange={handleChange}
            id="two"
            inline
            label="2 пересадки"
            type="checkbox"
            checked={state.two}
          />
          <SearchInput
            name="three"
            onChange={handleChange}
            id="three"
            inline
            label="3 пересадки"
            type="checkbox"
            checked={state.three}
          />
        </div>
      </StyledForm>
    </Router>
  );
};

export default Search;
