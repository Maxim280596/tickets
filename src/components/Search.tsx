import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';
import { Formik } from 'formik';

import { filterTickets } from '../redux/ducks/ticketsReduser';
import { StyledForm } from '../styled/StyledSearch';
import { SearchInput } from './SearchInput';
import { checkboxOptions } from '../constants';
import { Checkbox } from '../types';

const Search: React.FC = () => {
  const history = useHistory();
  const [options, setOptions] = useState(checkboxOptions);
  const dispatch = useDispatch();
  const toggleCheckbox = (option: { id: number }) => {
    const updatedOptions = options.map((item: Checkbox) => {
      if (item.id === option.id) {
        item.checked = !item.checked;
        dispatch(filterTickets(options));
        history.push(item.path);
      }
      if (item.checked && option.id !== item.id) {
        item.checked = false;
      }
      return item;
    });
    setOptions(updatedOptions);
  };

  return (
    <Router>
      <div>
        <Formik initialValues={checkboxOptions} onSubmit={() => {}}>
          {(formik: {}) => (
            <StyledForm>
              <h5>КОЛИЧЕСТВО ПЕРЕСАДОК</h5>
              <SearchInput
                name="checkboxOptions"
                options={checkboxOptions}
                toggleCheckbox={toggleCheckbox}
              />
            </StyledForm>
          )}
        </Formik>
      </div>
    </Router>
  );
};

export default Search;
