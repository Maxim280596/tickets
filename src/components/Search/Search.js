import React, { useState } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { useHistory } from 'react-router-dom';

import { StyledForm } from '../../styled/Search/StyledSearch';
import { SearchInput } from '../SearchInput/SearchInput';

export const Search = () => {

  const [state, setState] = useState({
    all: false,
    no: false,
    one: false,
    two: false,
    three: false,
  });

  let history = useHistory();

  return (
    <Router>
      <StyledForm>
        <div>
          <h5>КОЛИЧЕСТВО ПЕРЕСАДОК</h5>
          <SearchInput
            name={1}
            onChange={(e) => {
              history.push(`/${e.target.id}`);
              setState({ all: !e.target.checked });
              console.log(state)
            }}
            id="all"
            inline
            label="Все"
            type="checkbox"
            checked={state.id}
          />
          <SearchInput
            name={2}
            onChange={(e) => {
              history.push(`/${e.target.id}`);
              setState({ no: !e.target.checked });
              console.log(state)
            }}
            id="no"
            inline
            label="Без пересадок"
            type="checkbox"
            checked={state.id}
          />
          <SearchInput
            name={3}
            onChange={(e) => {
              history.push(`/${e.target.id}`);
              setState({ one: !e.target.checked });
              console.log(state)
            }}
            id="one"
            inline
            label="1 пересадка"
            type="checkbox"
            checked={state.id}
          />
          <SearchInput
            name={4}
            onChange={(e) => {
              history.push(`/${e.target.id}`);
              setState({ two: !e.target.checked });
              console.log(state)
            }}
            id="two"
            inline
            label="2 пересадки"
            type="checkbox"
            checked={state.id}
          />
          <SearchInput
            name={5}
            onChange={(e) => {
              history.push(`/${e.target.id}`);
              setState({ three: !e.target.checked });
              console.log(state)
            }}
            id="three"
            inline
            label="3 пересадки"
            type="checkbox"
            checked={state.id}
          />
        </div>
        
      </StyledForm>
    </Router>
    
  );
};
