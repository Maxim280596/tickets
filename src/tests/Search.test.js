import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Search from '../components/Search';

describe('Testing <Search/>', () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store;
  let search;
  beforeAll(() => {
    store = mockStore(initialState);
    search = render(
      <Provider store={store}>
        <Search />
      </Provider>
    );
  });
  it('should Search have be rendered correctly', () => {
    expect(toJson(search)).toMatchSnapshot();
  });
});
