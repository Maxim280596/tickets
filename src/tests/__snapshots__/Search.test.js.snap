// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Search/> should Search have be rendered correctly 1`] = `
<div>
  <form
    class="sc-bdfBwQ hxMzZH"
  >
    <h5>
      КОЛИЧЕСТВО ПЕРЕСАДОК
    </h5>
    <div>
      <div
        class="sc-gsTCUz btRBl form-check"
      >
        <input
          checked=""
          class="form-check-input"
          length="10"
          type="checkbox"
          value="Все"
        />
        <label
          class="form-check-label"
          title=""
        >
          Все
        </label>
      </div>
      <div
        class="sc-gsTCUz btRBl form-check"
      >
        <input
          class="form-check-input"
          length="0"
          type="checkbox"
          value="Без пересадки"
        />
        <label
          class="form-check-label"
          title=""
        >
          Без пересадки
        </label>
      </div>
      <div
        class="sc-gsTCUz btRBl form-check"
      >
        <input
          class="form-check-input"
          length="1"
          type="checkbox"
          value="1 пересадка"
        />
        <label
          class="form-check-label"
          title=""
        >
          1 пересадка
        </label>
      </div>
      <div
        class="sc-gsTCUz btRBl form-check"
      >
        <input
          class="form-check-input"
          length="2"
          type="checkbox"
          value="2 пересадки"
        />
        <label
          class="form-check-label"
          title=""
        >
          2 пересадки
        </label>
      </div>
      <div
        class="sc-gsTCUz btRBl form-check"
      >
        <input
          class="form-check-input"
          length="3"
          type="checkbox"
          value="3 пересадки"
        />
        <label
          class="form-check-label"
          title=""
        >
          3 пересадки
        </label>
      </div>
    </div>
  </form>
</div>
`;
